system-contracts/contracts/erc4337/EntryPointV01.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IEntryPoint, PackedUserOperation} from "../interfaces/IEntryPoint.sol";
+import {IBootloaderUtilities} from "../interfaces/IBootloaderUtilities.sol";
+import {IAccount} from "../interfaces/IAccount.sol";
+import {IAccountCodeStorage} from "../interfaces/IAccountCodeStorage.sol";
+import {INonceHolder} from "../interfaces/INonceHolder.sol";
+import {IContractDeployer} from "../interfaces/IContractDeployer.sol";
+import {ISystemContext} from "../interfaces/ISystemContext.sol";
+import {ContractDeployer} from "../ContractDeployer.sol";
+import {Transaction, TransactionHelper, EIP_712_TX_TYPE, LEGACY_TX_TYPE, EIP_2930_TX_TYPE, EIP_1559_TX_TYPE} from "../libraries/TransactionHelper.sol";
+import {BOOTLOADER_FORMAL_ADDRESS, SYSTEM_CONTEXT_CONTRACT, NONCE_HOLDER_SYSTEM_CONTRACT, ACCOUNT_CODE_STORAGE_SYSTEM_CONTRACT, DEPLOYER_SYSTEM_CONTRACT} from "../Constants.sol";
+import {SystemContractBase} from "../abstract/SystemContractBase.sol";
+import {SystemContext} from "../SystemContext.sol";
+import {RLPEncoder} from "../libraries/RLPEncoder.sol";
+import {EfficientCall} from "../libraries/EfficientCall.sol";
+import {UnsupportedTxType, InvalidSig, SigField, HashMismatch} from "../SystemContractErrors.sol";
+
+import {SystemContractsCaller, CalldataForwardingMode} from "../libraries/SystemContractsCaller.sol";
+
+/**
+ * @author Matter Labs
+ * @custom:security-contact security@matterlabs.dev
+ * @notice An EIP4337 EntryPoint contract implementation
+ * built on top of the native ZKsync Account Abstraction.
+ */
+contract EntryPointV01 is IEntryPoint, SystemContractBase {
+    using TransactionHelper for *;
+
+    function handleUserOps(PackedUserOperation[] calldata _ops) external {
```

Comment:
If I understand correctly the purpose of this contract is to ensure that anyone can take any EIP712 transaction and run it through this contract.

- Not sure what about refunds in this case, whether the contract wont be overcharged.
- If zksync-ethers calculated tx hashes on its own, they no longer be valid
- Generally will have to cross check through our AA community what features are used for custom accounts